src/backend.rs

```diff
@@ -401,11 +406,18 @@ impl State {
         Ok(adapter)
     }
 
-    fn with_services<F>(&self, selectors: Vec<ServiceSelector>, mut cb: F) where F: FnMut(&Arc<SubCell<ServiceData>>) {
+    fn with_services<F>(&self, selectors: Vec<ServiceSelector>, mut cb: F)
+        where F: FnMut(&Arc<SubCell<ServiceData>>, &Option<TagStorage>) {
+
+        let store = match self.db_path {
```

Comment:
Does this mean that we are opening the database each time we look at the list of services? That doesn't sound good.